### DISCUSSION TITLE
Question: Is this multimodule compatabile?

### DISCUSSION THREAD

**Discussion by Daniel-Yonkov** [wiverson/maven-jpackage-template#43] [General]
Hello,
I am considering using this template for building a demo application I am building on Java 16 using JavaFX as a separate UI module and a Logic as another. 

@wiverson - Would you advice if this is a suitable approach to JVM integration using shader plugin and multimodule project?

Thank you for your time and effort on this template :+1:

**Comment by wiverson** [comment]
Off the cuff, I would say that if you want to build a JAR that has your logic and another JAR that has just your JavaFX ui-specific code with Maven multi-module, that would probably work fine. Effectively, you are just declaring that the UI code has an ordinary Maven dependency on the logic, and Maven multi-module is just helping with the build process a bit.

I would suggest avoiding using the shade plugin for JavaFX apps, as you really don't need it. JARs have a bit more complexity (e.g. service and manifest declarations, multi-version, module-info files, etc) that make shading less useful.

Just to make sure everything is as confusing as possible, there are Maven multi-module projects, and then there is the Java Module System. Very, very different things.
